server/src/main/java/org/candlepin/liquibase/PerOrgProductsMigrationTask.java

```diff
@@ -0,0 +1,455 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.liquibase;
+
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+
+/**
+ * The PerOrgProductsMigrationTask performs the post-db upgrade data migration to the cp2_* tables.
+ */
+public class PerOrgProductsMigrationTask extends LiquibaseCustomTask {
+
+    protected Map<String, String> migratedProducts;
+    protected Map<String, String> migratedContent;
+    private Timestamp migrationTime;
+
+    public PerOrgProductsMigrationTask(Database database, CustomTaskLogger logger) {
+        super(database, logger);
+
+        this.migratedProducts = new HashMap<String, String>();
+        this.migratedContent = new HashMap<String, String>();
+        this.migrationTime = new Timestamp(System.currentTimeMillis());
+    }
+
+    /**
+     * Generates a prepared statement for performing a bulk insert into the given table
+     *
+     * @param table
+     *  the table into which to insert the data
+     *
+     * @param rows
+     *  the number of rows being inserted
+     *
+     * @param cols
+     *  the columns receiving data for each row
+     *
+     * @return
+     *  a PreparedStatement instance representing the bulk insert operation
+     */
+    protected PreparedStatement generateBulkInsertStatement(String table, int rows, String... cols)
+        throws DatabaseException, SQLException {
+
+        if (rows > 0) {
+            StringBuilder builder, rowbuilder;
+
+            if (!this.database.getDatabaseProductName().matches(".*(?i:oracle).*")) {
+                rowbuilder = new StringBuilder(2 + cols.length << 1);
```

Comment:
It's a little odd to see the shift operator here.  Is the goal to double `cols.length` and add 2?  If so, I feel like "2 + cols.length * 2" is going to be a lot clearer to future maintainers and the performance cost is probably negligible given that this code only runs once.